### PR TITLE
Add DID resolver support

### PR DESCRIPTION
**Pull request by PaulLeCam**

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+MIT License
```
Comment (oed) [inline LICENSE:1]: Thanks for adding this!

**Comment by PaulLeCam** [comment]
Yes good point, I'll add way to set the provider separately.
About the Resolver instance or registry it accepts both ways, cf https://github.com/ceramicnetwork/js-did/pull/4/files#diff-04c6e90faac2675aa89e2176d2eec7d8R93

**Comment by PaulLeCam** [comment]
@oed I made the change, if you can please have a look?
It should work in the same way as it's being implemented in js-idx.

File: src/index.ts [hunk 2]
```diff
@@ -41,37 +41,41 @@ export interface DIDOptions {
...
       throw new Error('DID is not authenticated')
     }
     return this._did
   }
 
-  public setProvider(provider: DIDProvider): void {
+  setProvider(provider: DIDProvider): void {
     if (this._client?.connection !== provider) {
       this._client = new RPCClient(provider)
       this._did = undefined
```
[the remark this answers is not in the document]
Comment (PaulLeCam) [reply src/index.ts]: Yes I'll make the change 👍

**Pull request merged**